### ISSUE TITLE
Cannot run master

### ISSUE THREAD

**Issue by HectorMM3** [RuminantFarmSystems/RuFaS#76]
I attempted to run Master as is and I get these errors. 

Simulating: barnyard.json
Traceback (most recent call last):
  File "C:/Users/Hector.Menendez/OneDrive - USDA/Desktop/2020/RuFaS/MASM/main.py", line 47, in <module>
    main()
  File "C:/Users/Hector.Menendez/OneDrive - USDA/Desktop/2020/RuFaS/MASM/main.py", line 40, in main
    RUFAS.simulate(input_file_path)
  File "C:\Users\Hector.Menendez\OneDrive - USDA\Desktop\2020\RuFaS\MASM\RUFAS\simulation_engine.py", line 65, in simulate
    annual_simulation()
  File "C:\Users\Hector.Menendez\OneDrive - USDA\Desktop\2020\RuFaS\MASM\RUFAS\simulation_engine.py", line 117, in annual_simulation
    daily_simulation()
  File "C:\Users\Hector.Menendez\OneDrive - USDA\Desktop\2020\RuFaS\MASM\RUFAS\simulation_engine.py", line 84, in daily_simulation
    routines.daily_soil_routine(state.soil, state.crop, weather, time)
  File "C:\Users\Hector.Menendez\OneDrive - USDA\Desktop\2020\RuFaS\MASM\RUFAS\routines\soil\soil.py", line 189, in daily_soil_routine
    phosphorus_cycling.update_all(soil, weather, time)
  File "C:\Users\Hector.Menendez\OneDrive - USDA\Desktop\2020\RuFaS\MASM\RUFAS\routines\soil\phosphorus_cycling\phosphorus_cycling.py", line 31, in update_all
    soluble_P.update_all(soil)
  File "C:\Users\Hector.Menendez\OneDrive - USDA\Desktop\2020\RuFaS\MASM\RUFAS\routines\soil\phosphorus_cycling\soluble_P.py", line 28, in update_all
    layer.DRP_leachate = min(40.0, exp((layer.soil_P * 1.5 - layer.iso_inter / layer.iso_slope)))
OverflowError: math range error

**Comment by wmaxdonovan** [comment]
@HectorMM3 This math overflow error was originally thought to be a case of mistaken parentheses but now seems to arise from *appropriate* model function (or some inappropriate model function arising elsewhere) when running Alfalfa. A method to avoid the error is being introduced with dj_fields but does not get at the root of the problem.

**Comment by wmaxdonovan** [comment]
> This math overflow error was originally thought to be a case of mistaken parentheses but now seems to arise from _appropriate_ model function (or some inappropriate model function arising elsewhere) when running Alfalfa. A method to avoid the error is being introduced with dj_fields but does not get at the root of the problem.

@Jacob8399 the comment about alfalfa causing the error was based on a comment left in the dj_fields pull request, but running on master this does not seem to be the case. Any insight?

**Comment by wmaxdonovan** [comment]
If you need to run master prior to the merge completing, you can copy this code in place of the code referenced in the last line of the traceback to circumvent the problem. The comment is included for further explanation

**Jacob just updated this with the new version below**

# S.5/6.E.3
        # this if statement avoids a range error that can happen within the exp()
        if ((layer.soil_P * 1.5 - layer.iso_inter) / layer.iso_slope) <= log(40):
            layer.DRP_leachate = min(40.0, exp((layer.soil_P * 1.5 - layer.iso_inter) / layer.iso_slope))
        else:
            layer.DRP_leachate = 40

**Comment by Jacob8399** [comment]
There is updated code for it in dj_fields that is cleaner that I did today. It was a random error I came across while messing with alfalfa default values during that testing process. At certain levels of PHU, the labile_P for alfalfa would get high enough that the value inside the exp() would be higher than what python can handle. It shouldn't happen when alfalfa is run but instead when certain parts of alfalfa are changed. The if statement I added should catch that error and will soon be in master once dj_fields is merged. This error being caught does not mess with model functionality, it just keeps an equation from running at certain, but correct, times.
